tests/FunctionalTestCase.php

```diff
@@ -278,7 +279,7 @@ protected function getServerVersion(ReadPreference $readPreference = null)
         $document = current($cursor->toArray());
 
         if (isset($document['version']) && is_string($document['version'])) {
-            return $document['version'];
+            return preg_replace('#^(\d+\.\d+\.\d+).*$#', '\1', $document['version']);
```

Comment:
This change was necessary since my tests were being skipped. My local MongoDB server reported `4.9.0-alpha4-...` as version number, which `version_compare` interprets as being lower than `4.9.0` (due to the alpha flag). The [Version String in the unified test format](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#version-string) makes no mention of stabilities, but instead requires the version to conform to the `<major>.<minor>.<patch>` schema.

Reply:
This seems related to https://github.com/mongodb/specifications/pull/931.

Works for me. The other alternative would be to use `versionArray` and build the string ourselves. Presumably, we'll need to do this for the test runners as well?

Reply:
Yes, that issue was raised afterwards, and we have PHPLIB-634 as a downstream ticket to take care of this. If you don't mind I'd leave this in for now (as it ensures that the tests run as expected) and then the logic for all test runners as part of PHPLIB-634.